### ISSUE TITLE
bug: Editor Crash when Input Variable is a Guild Variable

### ISSUE THREAD

**Issue by RecoStar** [AscensionGameDev/Intersect-Engine#1455]
### Description

The editor will crash when you try to slect "Guild Variable" from with in a "Input Variable" on a map event.

### Steps to Reproduce

1. Create a Guild Variable
2. Add a map Event
3. Add input variable
4. Select "Guild Variable"
5. Editor Crash

### Version with bug

0.7.1.49-beta Full

### Last version that worked well

Not sure

### Affected platforms

Windows

### Did you find any workaround?

No

### Relevant log output

```shell
Intersect Editor and errors-Intersect Editor.log:

2022-09-20 18:15:06.658 [Error] IndexOutOfRangeException: Index was outside the bounds of the array.
    Stack:    at System.ComponentModel.ReflectPropertyDescriptor.SetValue(Object component, Object value)
   at System.Windows.Forms.RadioButton.PerformAutoUpdates(Boolean tabbedInto)
   at System.Windows.Forms.RadioButton.set_Checked(Boolean value)
   at System.Windows.Forms.RadioButton.OnClick(EventArgs e)
   at System.Windows.Forms.RadioButton.OnMouseUp(MouseEventArgs mevent)
   at DarkUI.Controls.DarkRadioButton.OnMouseUp(MouseEventArgs e)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

### Duplicate Bug Check

- [X] This bug report is not a duplicate to the best of my knowledge.

**Comment by RecoStar** [comment]
Tested other versions and they seem to be working fine. 

Last build this issue seems to have came from?

**Comment by RecoStar** [comment]
Further investigation. This only happens when guild variables are in the system.

**Comment by Kamusvalenzuela** [comment]
> Tested other versions and they seem to be affected. Tested down to version 0.7.1.47

> Further investigation. This only happens when guild variables are in the system.

i can't reproduce following those steps (Tested on 0.7.1.48)
Tested with new and old guild variables and works well for me

S.O Windows 10

[Video of example](https://www.ascensiongamedev.com/resources/filehost/508aa0ddbdee945f6e8c17a7a1c98ba0.mp4)

**Comment by RecoStar** [comment]
Thanks for testing. Strange as this is not working at all for me. I have just downloaded a fresh copy of intersect and tried and again same issue.

I am using windows 10, but it shouldn't be OS related?

I don't understand what is happening. No matter what I do, if I create a Guild Variable and try to use Input Variable and select a Guild Variable the editor crashes.

I have tested with different variable types, different wording, spacing in wording, anything to try and pin point the issue more.

**Comment by lodicolo** [comment]
This is happening totally outside of our code, the crash is coming from inside of the WinForms library itself.

My guess is that you two are not running with the exact same version of some dependency (be it WinForms or .NET), but I am not sure how to thoroughly diagnose this.

@RecoStar Can you

1. open System Information (part of Windows) and provide a screenshot of this: on your machine
![image](https://user-images.githubusercontent.com/1476550/192110491-45744ef1-0035-410e-b107-5c848c62a696.png)
2. tell us what version of .NET Framework do you have installed?

**Comment by RecoStar** [comment]
Strange as I don't think I have made any changes to .NET in a while.
1.
![image](https://user-images.githubusercontent.com/25243860/192115361-a118d10d-7ac0-484f-bad6-94e240e6539f.png)

2. 4.7.2
![image](https://user-images.githubusercontent.com/25243860/192115404-44e697ce-677e-47c6-981e-f8e3d383e9e0.png)

I will see if updating .NET and such will fix this issue.

**Comment by RecoStar** [comment]
Just updated .NET Framework to latest version 4.8 and still getting same issue.

**Comment by Arufonsu** [comment]
> .NET Framework to latest version 4.8

Intersect uses .NET Framework 4.6.2 right now, have you tried with that?

**Comment by lodicolo** [comment]
I am also unable to reproduce this (not surprised).

I do think that this has to be an issue with the version of .NET Framework you have. I would install the runtime for .NET Framework 4.6.2, and if still crashes also install the runtime for 4.6.1.

**Comment by RecoStar** [comment]
I hate .NET framework. Couldn't install 4.6.2 or 4.6.1 because a newer version is installed. Uninstalled .NET going backwards in versions, cleared all .NET and it still says its install lol. Going to have to sort it out so might take a bit.

Dont understand though why only one part of the editor was crashing due to my .NET version, its always worked before.

**Comment by RecoStar** [comment]
Ok that was a pain and took some time, but I completed removed .NET Framework from my machine and reinstalled starting from 4.6 all the way to latest version. 

Issue is still happening, so cant be .NET Framework surely?

![image](https://user-images.githubusercontent.com/25243860/192143740-1c15b014-eb39-4994-b953-ff7cd1cd36c6.png)

**Comment by lodicolo** [comment]
> Dont understand though why only one part of the editor was crashing due to my .NET version, its always worked before.

That part of your editor has worked in the past? It only just started causing problems recently?

**Comment by RecoStar** [comment]
It has always worked, never had this issue before. I was adding Input Variables with Guild Variables a couple weeks ago.

**Comment by lodicolo** [comment]
Are there additional logs/can you record this in a video clip please?

**Comment by RecoStar** [comment]
I will sort out a video later today after work. Thanks.

**Comment by WeylonSantana** [comment]
Download OCAM - https://ohsoft.net/eng/ocam/download.php?cate=1002
It's a paid program, but you can use it for free, you choose the recording area, then you can put it on a normal screen size, and decrease the recording quality to make the final file smaller.
I can record up to 2 min at 1920x1080, without the file exceeding 8mb.

Try recording with this program and let us know if it worked

**Comment by RecoStar** [comment]
https://user-images.githubusercontent.com/25243860/192795603-483dd4f6-9bf5-4c55-a610-04c3c8b2b850.mp4

**Comment by RecoStar** [comment]
Someone else has reported this same issue on forum topic for this issue.

**Comment by WeylonSantana** [comment]
This is an intersection bug verified, cant input guild variable
![image](https://user-images.githubusercontent.com/63019821/192863169-d57904b4-7a9e-4d99-869a-4a446b26db97.png)

**Comment by lodicolo** [comment]
> This is an intersect bug verified, cant input guild variable

Correction, @WeylonSantana @RecoStar, there is another definite bug in Intersect that is related and may be the cause. We do _not_ have sufficient evidence to say they are the same until Weylon's PR is merged and RecoStar can test it. I _suspect_ it might be the same bug, but the two bugs do not have any matching stack traces -- only a matching exception.

**Comment by lodicolo** [comment]
@RecoStar Please test to see if you are able to reproduce your bug in 0.7.1.57 which is [now available](https://www.ascensiongamedev.com/topic/4232-main-builds/?do=getLastComment)

**Comment by lodicolo** [comment]
So Weylon hadn't mentioned to me that he actually did have logs that matched, but he actually does have identical logs to yours. This should be safe to close as fixed in 0.7.1.57.

**Comment by RecoStar** [comment]
I can confirm that Weylons compiled version worked for. 

Just tested 0.7.1.57 and issue looks to be resolved :)

Good job guys, and thanks.
